### DISCUSSION TITLE
Migration to modern decorator

### DISCUSSION THREAD

**Discussion by kk8787** [MichalLytek/type-graphql#1549] [Q&A]
Nice to meet you, thank you for the wonderful library.

Now that typescript 5.2 has been released and decorator metadata has reached stage 3, are there any plans to support this in typegraphql? Also, when is it likely to happen?
I'm considering migrating from nestjs/graphql, but the only thing that bothers me is the dependency on legacy decorator.
By the way, other options are nexus and pothos, but I felt that nexus is not actively developed and pothos is complicated and difficult to debug.

**Comment by MichalLytek** [comment]
See https://github.com/MichalLytek/type-graphql/issues/1411
